### DISCUSSION TITLE
RNFBMessagingModule.getToken got 4 arguments, expected 2 error

### DISCUSSION THREAD

**Discussion by Canser** [invertase/react-native-firebase#5609] [Q&A] [answered]
Android app works in debug mode but crashes at the start when I build a release version. IOS app works fine.

I debugged using the adb logcat command and have caught:
```

08-18 11:28:44.144 13322 13360 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
08-18 11:28:44.144 13322 13360 E AndroidRuntime: Process: com.example.example, PID: 13322
08-18 11:28:44.144 13322 13360 E AndroidRuntime: com.facebook.react.bridge.NativeArgumentsParseException: RNFBMessagingModule.getToken got 4 arguments, expected 2
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:349)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at android.os.Handler.handleCallback(Handler.java:809)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at android.os.Handler.dispatchMessage(Handler.java:102)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at android.os.Looper.loop(Looper.java:166)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
08-18 11:28:44.144 13322 13360 E AndroidRuntime:    at java.lang.Thread.run(Thread.java:784)

```
In short, the error message is: RNFBMessagingModule.getToken got 4 arguments, expected 2

I am using the getToken method as seen below:

```
useEffect(() => {
    if (isLoggedIn) {
      messaging()
        .getToken()
        .then((token) => {
          if (
            isLoggedIn &&
            (!user.device_token?.length || user.device_token !== token)
          ) {
            updateProfile({
              deviceToken: token,
            });
          }
        });
    }
  }, [isLoggedIn]);
```
Updated the rn-firebase packages to the latest but still did not work. I am completely lost at the moment.

Do you have any suggestions?

**Comment by mikehardy** [comment, marked as answer]
That is usually what happens when native code and javascript bundles are the wrong versions.
Like, you have native react-native-firebase code built for 12.6.0 but react-native-firebase javascript from 12.7.x
I would make very very very sure that your build system is not incorrectly caching the output of some build step, in a fully clean build this should not be possible.

**Comment by Canser** [comment]
IMPORTANT EDIT:

Okay, I just realised what you meant. Before running the assembleRelease; first I ran below and it worked!  Thank you.

`react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res/`

-------------------------------

@mikehardy Sorry for the late reply. I am not sure that I understand what you meant. How can I be sure about what you suggest?

Found a topic about cleaning the cache(?) [here](https://gist.github.com/jarretmoses/c2e4786fd342b3444f3bc6beff32098d).

Could you please tell me if any other way to do this?

--- Edit ---

I ran this command but did not work for me.

`watchman watch-del-all && rm -rf node_modules && npm install && rm -rf /tmp/metro-* && npm run start --reset-cache`

Also ran; `./gradlew clean`.

And one more thing; I am generating the APK using the  `./gradlew assembleRelease` command.

**Reply by mikehardy** [reply]
Usually what I do is `npx react-native-clean-project`, for my release builds I use the command line options to selectively turn on (or off) different types of cache clearing so that it has the right balance of "starts pretty clean" but "won't take forever" (like if you delete your system pod cache). Either way, glad it's working for you
